### PR TITLE
Add FixedTuple member enforcing a given number of items

### PR DESCRIPTION
**Pull request by MatthieuDartiailh**

### CODE REVIEW

**Comment by MatthieuDartiailh** [comment]
@frmdstryr do you have an interest in this feature and do you want to review ?

File: atom/src/validatebehavior.cpp [hunk 1]
```diff
@@ -463,6 +482,56 @@ tuple_handler( Member* member, CAtom* atom, PyObject* oldvalue, PyObject* newval
...
+{
+    if( !PyTuple_Check( newvalue ) )
+    {
+        return validate_type_fail( member, atom, newvalue, "tuple" );
+    }
+    cppy::ptr tupleptr( cppy::incref( newvalue ) );
+
+    // Create a copy in which to store the validated values
+    Py_ssize_t size = PyTuple_GET_SIZE( newvalue );
+    cppy::ptr tuplecopy = PyTuple_New( size );
```
Comment (frmdstryr) [inline atom/src/validatebehavior.cpp:496]: This could be moved to after the size check

**Comment by MatthieuDartiailh** [comment]
That's a good point. I will look into it. The error message may indeed be more challenging but I will look into it.

**Comment by MatthieuDartiailh** [comment]
So the issue with re-using the same member is that it is not backward compatible. Changes to how we interpret annotations makes the system stricter but it bothers me less since it should not break anything that was previously valid.

**Comment by frmdstryr** [comment]
I am normally against breaking backwards compatibility but in this case I think it is worth it. I'm so used to using the `tuple[x, y, z]` with typing I actually found an invalid use of the current `Tuple` that passes what should be the second type as the default value. Maybe others can chime in but do whatever.

**Comment by MatthieuDartiailh** [comment]
Thanks @frmdstryr I will keep two separate members. 

Please note that if you use annotations instead of the actual members `tuple[x, y, z]` will use `FixedTuple` which should fix your issue.

**Comment by MatthieuDartiailh** [comment]
I forgot to address your comments (forgot to re-read the whole discussion). I will come back to it once I am done with #210

**Pull request merged**